packages/network/src/chart/Node.jsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { nodeStyleShape } from '../utils/propShapes';
+
+const proptypes = {
+  nodeStyles: nodeStyleShape,
+  node: PropTypes.object.isRequired,
+  onMouseMove: PropTypes.func,
+  onClick: PropTypes.func,
+  onMouseLeave: PropTypes.func,
+  onMouseEnter: PropTypes.func,
+};
+
+const defaultProps = {
+  nodeStyles: {
+    fill: '#15aabf',
+    stroke: 'none',
+    strokeWidth: 0,
+    opacity: 0.5,
+    defaultSize: 3,
+  },
+  onMouseMove: undefined,
+  onClick: undefined,
+  onMouseLeave: undefined,
+  onMouseEnter: undefined,
+};
+
+export default function Node(props) {
+  const { nodeStyles, node, onMouseMove, onClick, onMouseLeave, onMouseEnter } = props;
+  const { stroke, strokeWidth, fill, opacity, defaultSize } = nodeStyles;
+  return (
+    <g opacity={node.opacity || opacity} >
+      <circle
+        // key={node.id}
+        r={node.size || defaultSize}
+        fill={node.fill || fill}
+        stroke={node.stroke || stroke}
+        strokeWidth={strokeWidth}
+        onMouseMove={onMouseMove && ((event) => {
+          onMouseMove({ event, data: node });
+        })}
+        onMouseLeave={onMouseLeave && ((event) => {
+          onMouseLeave({ event, index: node.index });
+        })}
+        onMouseEnter={onMouseEnter && ((event) => {
```

Comment:
not sure we need `onMouseEnter` if we have `onMouseMove`?

Reply:
This is for highlighting nodes when hovering happened. onMouseEnter will be only triggered once at the very beginning when the mouse enters the node. However, onMouseMove will be triggered every time the mouse moves.

Reply:
oh I see, that makes sense, thanks for more context 👍